### ISSUE TITLE
Problem with embedding model 

### ISSUE THREAD

**Issue by npk7264** [langchain-ai/langchain#9486]
### Issue you'd like to raise.

I use embedding model from huggingface vinai/phobert-base:
![image](https://github.com/langchain-ai/langchain/assets/90046327/a5fef433-ae15-4d30-abf0-0fe14f1ca7db)
Then it has this problem:
WARNING:sentence_transformers.SentenceTransformer:No sentence-transformers model found with name /root/.cache/torch/sentence_transformers/vinai_phobert-base. Creating a new one with MEAN pooling.
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.

**Comment by npk7264** [comment]
@dosu-beta how will I solve this problem?

**Comment by npk7264** [comment]
@dosu-bot Can you tell me which task of Huggingface Model is suitable for Retrieval in LangChain?

**Comment by npk7264** [comment]
@dosu-bot I am mentioning about tasks: sentence similarity, fill mask, token classification,... When I use HuggingFaceEmbeddings, which task is suitable to fix error like when I use vinai/phobert-base?

**Comment by npk7264** [comment]
@dosu-bot Tell me more detail about option 2, I want to modify the LangChain Python framework to support this model and the tasks "sentence-similarity"

**Comment by npk7264** [comment]
vinai/phobert-base is word-embedding model, while HuggingFacesEmbeddings in LangChain require sentence-embedding model, so it doesn't work. Am I thinking right?

**Comment by npk7264** [comment]
@dosu-bot vinai/phobert-base is word-embedding model, while HuggingFacesEmbeddings in LangChain require sentence-embedding model, so it doesn't work. Am I thinking right?

**Comment by npk7264** [comment]
@dosu-bot When I use custom PhoBertEmbedding():
hf_embedding = PhoBertEmbeddings()
vectordb = Chroma.from_documents(
    documents=data,
    embedding=hf_embedding,
    persist_directory=persist_directory
)

This is problem and how to fix them, give me suggest:
Token indices sequence length is longer than the specified maximum sequence length for this model (555 > 256). Running this sequence through the model will result in indexing errors

**Comment by npk7264** [comment]
@dosu-bot When I use custom PhoBertEmbedding():
hf_embedding = PhoBertEmbeddings()
vectordb = Chroma.from_documents(
documents=data,
embedding=hf_embedding,
persist_directory=persist_directory
)

This is two problem and how to fix them, give me suggest:

Token indices sequence length is longer than the specified maximum sequence length for this model (555 > 256). Running this sequence through the model will result in indexing errors
index out of range in self
